tests/unit/utils/test_wait_for_service.py

```diff
@@ -0,0 +1,22 @@
+from src.utils.wait_for_service import wait_for_service
+import pytest
+import logging
+import time
+
+
+def test_init_config_invalid_config_url(caplog):
+    search2_logger = logging.getLogger('search2')
+    # This ensures that logs are propagated and can be captured by caplog
+    search2_logger.propagate = True
```

Comment:
I don't think I had seen this option before.  I need to read up on it.

Reply:
Yeah. search_api2 turns propagation off in order to suppress duplicate log messages. There are apparently other methods for achieving that, like disabling handing of the top level logger and only ever logging to specific loggers or conversely only handling the top level logger, but didn't think it worth that effort just to be able to capture logs in the tests.